### PR TITLE
revert: match implementation with design goals

### PR DESCRIPTION
**Pull request by jchampio**
Continue to evolve and clean up the revert implementation. Each commit is self-contained.

### CODE REVIEW

File: hub/revert.go [hunk 1]
```diff
@@ -122,26 +89,31 @@ func (s *Server) Revert(_ *idl.RevertRequest, stream idl.CliToHub_RevertServer)
...
-		oldDir, err := utils.GetLogDir()
-		if err != nil {
-			return err
-		}
-		archiveDir = filepath.Join(filepath.Dir(oldDir), upgrade.GetArchiveDirectoryName(s.UpgradeID, time.Now()))
+	if s.UseLinkMode {
+		// In any of the three cases of source cluster recovery in link-mode pg_upgrade, it is correct
+		// to restore the pg_control file.  See the "Reverting to old cluster" section of
+		// https://www.postgresql.org/docs/9.4/pgupgrade.html
+		st.Run(idl.Substep_RESTORE_PGCONTROL, func(streams step.OutStreams) error {
```
Comment (kalensk) [inline hub/revert.go:96]: minor nit: When I first read this I wasn't sure what "In any of the three cases" referred to. Maybe something like:

> In link-mode it is always correct to restore the pg_control file in any of the three cases of source cluster recovery as outlined in the "Reverting to old cluster" section of
https://www.postgresql.org/docs/9.4/pgupgrade.html

File: hub/revert.go [hunk 2]
```diff
@@ -69,42 +69,9 @@ func (s *Server) Revert(_ *idl.RevertRequest, stream idl.CliToHub_RevertServer)
...
-			// https://www.postgresql.org/docs/9.4/pgupgrade.html
-			st.Run(idl.Substep_RESTORE_PGCONTROL, func(streams step.OutStreams) error {
-				return RestoreMasterAndPrimariesPgControl(streams, s.agentConns, s.Source)
-			})
-		}
-	}
-
+	// The target cluster's data directories and tablespaces must be removed.
+	// TODO: combine the primary/master data directory deletion into a single
+	// substep implementation, to match the prevailing patterns.
```
Comment (kalensk) [inline hub/revert.go:74]: Would it be possible to do this soon, such as right after this PR or so? I would find that very useful for the long-term maintenance and readability of the code.

Comment (jchampio) [reply hub/revert.go:74]: Added a chore; it's a decent amount of unrelated churn.

File: hub/revert.go (hunk 1 above)
Comment (jchampio) [reply hub/revert.go]: Updated with additional context.

**Pull request merged**